File: src/main/java/cookiedragon/luchadora/util/IRenderable.java
package cookiedragon.luchadora.util;

/**
 * @author cookiedragon234 16/Dec/2019
 */
public interface IRenderable
{
	void render(Vec2f mousePos);
	boolean mouseClick(Vec2f mousePos, int mouseID);
	boolean mouseRelease(Vec2f mousePos, int mouseID);
	boolean mouseClickMove(Vec2f mousePos, int mouseID);
}

File: src/main/kotlin/cookiedragon/luchadora/util/IRenderable.kt
package cookiedragon.luchadora.util

/**
 * @author cookiedragon234 17/Feb/2020
 */
interface IRenderable {
	fun render(mousePos: Vec2f)
	fun mouseClick(mousePos: Vec2f, mouseID: Int): Boolean
	fun mouseRelease(mousePos: Vec2f, mouseID: Int): Boolean
	fun mouseClickMove(mousePos: Vec2f, mouseID: Int): Boolean
}
